services/QuillLMS/app/controllers/teachers/progress_reports/diagnostic_reports_controller.rb

```diff
@@ -192,15 +201,39 @@ def redirect_to_report_for_most_recent_activity_session_associated_with_activity
   end
   # rubocop:enable Metrics/CyclomaticComplexity
 
-  def assign_selected_packs
-    if params[:selections]
-      params[:selections].map { |s| s['id']}.each do |ut_id|
-        ut = UnitTemplate.find(ut_id)
-        Unit.unscoped.find_or_create_by(unit_template_id: ut.id, name: ut.name, user_id: current_user.id)
-      end
+  def assign_independent_practice_activity_packs
```

Comment:
Are we concerned about these two types of assignment (independent vs whole class) drifting from each other with regards to params, what is saved, run, etc.? I see in the frontend code, that these requests are sent independently (so this wouldn't require a refactor there).


I'm not against this approach and I don't see this as a blocker, just wondering aloud about the long-term ramifications of this (e.g. any assignment code must hit multiple independent endpoints to assign).

Reply:
I think based on how much logic is different between these two endpoints already these should be separate paths, but I think the general point is worth discussing with the team.